feat(replays): Render DOM Events as a list

danecando opened the pull request:
### Changes 
- Updates mutations table to a list view
- Adds border around list and makes list scrollable
- Adds click handler for timestamp to change current time
- Adds mouseEnter and mouseLeave handlers for highlighting

### Notes
I know we don't have an official mock for this yet so just throwing up what I put together for some initial feedback

Closes #36892 
Closes #37307 

![Screen Shot 2022-08-01 at 1 16 20 PM](https://user-images.githubusercontent.com/3721977/182206322-d8d1888e-d97a-4ee8-bff3-db8bc182b5b2.png)
![Screen Shot 2022-08-01 at 1 16 35 PM](https://user-images.githubusercontent.com/3721977/182206328-2784a731-4898-475c-8175-0d9a8ae9a0b1.png)


<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

danecando committed:
feat(replays): initial markup/styles for DOMEvents list

ryan953 commented on static/app/views/replays/detail/domMutations/index.tsx:
```diff
@@ -103,12 +115,23 @@ const Title = styled('span')`
+  max-width: 100%;
+`;
+
+const StepConnector = styled('div')`
```
looks like this `<div>` is going to be a child of the `<ul>?

danecando committed:
Merge branch 'master' into feat/replays/36892-dom-inspector

ryan953 commented on static/app/views/replays/detail/domMutations/index.tsx:
```diff
@@ -22,60 +20,67 @@ function DomMutations({replay}: Props) {
   const startTimestamp = replay.getEvent().startTimestamp;
 
+  if (isLoading) {
+    return null;
```
is this sufficient?

danecando replied:
Yeah the parent tab section already handles the loading state so that is there 'just in case'. It doesn't ever render when I was testing it

ryan953 commented on static/app/views/replays/detail/domMutations/index.tsx:
```diff
@@ -22,60 +20,67 @@ function DomMutations({replay}: Props) {
-const StyledPanelTable = styled(PanelTable)`
-  grid-template-columns: max-content max-content 1fr max-content;
-  font-size: ${p => p.theme.fontSizeSmall};
+const MutationList = styled('ul')`
+  list-style: none;
+  position: relative;
```
This should be a scrollable area

and i like the borders we have on the network table, for example, i think it'd be nice and consistent if we had that same border on this component (and eventually all of them within the focus area)

ryan953 replied:
it's gonna need to be it's own scrollable area because we're going to need to follow up and do some perf improvements. On replays with lots of clicks,.... it's a bit slow to render all the code blocks, so many are below the fold.

danecando replied:
@ryan953 is this what you had in mind?

![Screen Shot 2022-08-02 at 10 32 12 AM](https://user-images.githubusercontent.com/3721977/182400209-66504983-b923-480d-9e36-21d739817ad3.png)

ryan953 replied:
ya, i think that looks good! keeps everything together

ryan953 replied:
lets set `height: 100%` instead of `max-height` tho

danecando committed:
feat(replays): feedback changes and crumb handlers

- Adds border around list and makes list scrollable
- Adds click handler for timestamp to change current time
- Adds mouseEnter and mouseLeave handlers for highlighting

danecando commented:
@ryan953 I pushed updates from your feedback and also added the handlers from #37307

ryan953 commented on static/app/views/replays/detail/domMutations/index.tsx:
```diff
@@ -19,63 +20,126 @@ type Props = {
+        setCurrentHoverTime(relativeTimeInMs(item.timestamp ?? '', startTimestampMs));
+      }
+
+      if (item.data && 'nodeId' in item.data) {
+        // XXX: Kind of hacky, but mouseLeave does not fire if you move from a
+        // crumb to a tooltip
+        clearAllHighlights();
+        highlight({nodeId: item.data.nodeId, annotation: item.data.label});
+      }
```
you don't need to copy+paste this into here from breadcrumbs:

a) it's copy+paste, shouldn't be doing that
b) none of the console breadcrumbs that we render in this list will have `item.data.nodeId` values.

danecando committed:
fix(replays): removes highlighting feature and updates styles

danecando committed:
Merge branch 'master' into feat/replays/36892-dom-inspector

danecando committed:
fix(replays): max-height -> height

danecando committed:
fix(replays): style lint issue fix

billyvg commented on static/app/views/replays/detail/domMutations/index.tsx:
```diff
@@ -103,12 +162,23 @@ const Title = styled('span')`
+  height: 100%;
+  top: 28px;
+  left: 31px;
+  border-right: 1px ${p => p.theme.gray200} dashed;
```
```suggestion
  border-right: 1px ${p => p.theme.border} dashed;
```

billyvg commented:
@ryan953 I can't find the comments in GH's UI, but why don't these items have `nodeId`? I would expect them to have the same highlighting behavior to the breadcrumbs list, especially since they look very similar

danecando committed:
fix(replays): updated border color

ryan953 commented:
> @ryan953 I can't find the comments in GH's UI, but why don't these items have `nodeId`? I would expect them to have the same highlighting behavior to the breadcrumbs list, especially since they look very similar

I made the comment too quickly and was wrong. I was looking at the Console table and typing into here. :(
